dnn_rotate/nodes/rotate.py

```diff
@@ -0,0 +1,154 @@
+        self.image_width = rospy.get_param("~image_width", 410)
+        self.field_of_view = rospy.get_param("~field_of_view", 0.925)
+
+        # Setup the rotate service we serve
+        self.srv = rospy.Service("rotate", dnn_rotate.srv.String,
+                                 self.service_callback)
+        
+        # Create a proxy object for the move action server
+        self.move = actionlib.SimpleActionClient('move_base',
+                                                 move_base_msgs.msg.MoveBaseAction)
+        rospy.loginfo("Waiting for move service to be available")
+        if not self.move.wait_for_server(rospy.Duration(10.0)):
+           rospy.logerr("Move service not available")
+           return
+
+        # Create a proxy object fot the trigger service we call
+        rospy.loginfo("Waiting for detect service to be available")
+        try:
+            rospy.wait_for_service("/dnn_detect/detect", 10)
+            self.trigger = rospy.ServiceProxy("/dnn_detect/detect",
+                                              dnn_detect.srv.Detect)
+            rospy.loginfo("Rotate node ready")
+            self.ok = True
+        except:
+            rospy.logerr("detect service is not available") 
+
+    # This is called when we receive a rotate service call
+    def service_callback(self, rotate_req):
+        target = rotate_req.object.data
+        print("Received service call: rotate %s" % target)
+
+        # Create a response to our service
+        response = dnn_rotate.srv.StringResponse()
+
+        # Make a detection trigger service call
+        detect_req = dnn_detect.srv.DetectRequest()
+        detect_resp = self.trigger(detect_req)
+
+        # Look for matching objects
+        best_target = None
+        best_confidence = 0.0
+        for object in detect_resp.result.objects:
+            if object.class_name == target:
+                found_target = True
+                if object.confidence > best_confidence:
+                    best_target = object
+                    best_confidence = object.confidence
+
+        # Create a negative response to our rotate service call
+        if best_target is None: 
+            response.response.data = "No %s object found" % target
+            return response
+
+        # Calculate how far we need to rotate
+        x_center = (best_target.x_min + best_target.x_max) / 2.0
+        angle_per_pixel = self.field_of_view / self.image_width
+        angle_to_rotate = (self.image_width/2.0 - x_center) * angle_per_pixel
+        rospy.loginfo("Center of target %f, rotation required %f degrees" % \
+                      (x_center, degrees(angle_to_rotate)))
+        # The angle needs to be in quaternion form (a 3D angle)
+        q = tf.transformations.quaternion_from_euler(0, 0, angle_to_rotate)
+
+        # Create goal and actionlib call to rotate
+        goal = move_base_msgs.msg.MoveBaseGoal()
+        goal.target_pose.header.frame_id = "base_link"
```

Comment:
Does it make sense to be more 'correct' in this example code and transform the goal into odom? Since it would show a simple use of tf in python.

Plus that way works for other nav stacks as well.

On the other hand move_basic does provide this useful functionality, and a user should know that.

Reply:
Yes, we could add the code to transform the goal into the odom frame, and point out it is not necessary if move_basic is being used.